src/Microsoft.OpenApi/Extensions/OpenApiTypeMapper.cs

```diff
@@ -0,0 +1,85 @@
+    /// </summary>
+    public static class OpenApiTypeMapper
+    {
+        private static readonly Dictionary<Type, Func<OpenApiSchema>> _simpleTypeToOpenApiSchema = new()
+        {
+            [typeof(bool)] = () => new OpenApiSchema { Type = "boolean" },
+            [typeof(byte)] = () => new OpenApiSchema { Type = "string", Format = "byte" },
+            [typeof(int)] = () => new OpenApiSchema { Type = "integer", Format = "int32" },
+            [typeof(uint)] = () => new OpenApiSchema { Type = "integer", Format = "int32" },
+            [typeof(long)] = () => new OpenApiSchema { Type = "integer", Format = "int64" },
+            [typeof(ulong)] = () => new OpenApiSchema { Type = "integer", Format = "int64" },
+            [typeof(float)] = () => new OpenApiSchema { Type = "number", Format = "float" },
+            [typeof(double)] = () => new OpenApiSchema { Type = "number", Format = "double" },
+            [typeof(decimal)] = () => new OpenApiSchema { Type = "number", Format = "double" },
+            [typeof(DateTime)] = () => new OpenApiSchema { Type = "string", Format = "date-time" },
+            [typeof(DateTimeOffset)] = () => new OpenApiSchema { Type = "string", Format = "date-time" },
+            [typeof(Guid)] = () => new OpenApiSchema { Type = "string", Format = "uuid" },
+            [typeof(char)] = () => new OpenApiSchema { Type = "string" },
+            
+            // Nullable types
+            [typeof(bool?)] = () => new OpenApiSchema { Type = "boolean", Nullable = true },
+            [typeof(byte?)] = () => new OpenApiSchema { Type = "string", Format = "byte", Nullable = true },
+            [typeof(int?)] = () => new OpenApiSchema { Type = "integer", Format = "int32", Nullable = true },
+            [typeof(uint?)] = () => new OpenApiSchema { Type = "integer", Format = "int32", Nullable = true },
+            [typeof(long?)] = () => new OpenApiSchema { Type = "integer", Format = "int64", Nullable = true },
+            [typeof(ulong?)] = () => new OpenApiSchema { Type = "integer", Format = "int64", Nullable = true },
+            [typeof(float?)] = () => new OpenApiSchema { Type = "number", Format = "float", Nullable = true },
+            [typeof(double?)] = () => new OpenApiSchema { Type = "number", Format = "double", Nullable = true },
+            [typeof(decimal?)] = () => new OpenApiSchema { Type = "number", Format = "double", Nullable = true },
+            [typeof(DateTime?)] = () => new OpenApiSchema { Type = "string", Format = "date-time", Nullable = true },
+            [typeof(DateTimeOffset?)] = () => new OpenApiSchema { Type = "string", Format = "date-time", Nullable = true },
+            [typeof(Guid?)] = () => new OpenApiSchema { Type = "string", Format = "uuid", Nullable = true },
+            [typeof(char?)] = () => new OpenApiSchema { Type = "string", Nullable = true },
+            
+            [typeof(Uri)] = () => new OpenApiSchema { Type = "string" }, // Uri is treated as simple string
```

Comment:
Do we want to support `Uri?` too?

Reply:
And if we do I think the format needs to be specified

Reply:
The typeof operator cannot be used on `Uri?` since its a nullable reference type similar to `string?` and it does not have a standalone runtime representation https://github.com/dotnet/csharplang/discussions/3003?sort=old

Reply:
I had misread Safia's original comment. This is what I meant with mine.
https://json-schema.org/draft/2020-12/json-schema-validation.html#name-resource-identifiers
```suggestion
            [typeof(Uri)] = () => new OpenApiSchema { Type = "string", Format = "uri" }, // Uri is treated as simple string
```